cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/recommend-config.js

```diff
@@ -0,0 +1,403 @@
+                    <div class="price">${getPriceHtml(hit, html)}</div>
+                </div>
+            </div>
+        </div>
+    `;
+}
+
+/**
+ * This function is used to transform the item before it is displayed in the widget.
+ * It is used to add or modify properties of the item.
+ * @param {Object} item - Item to transform
+ * @returns {Object} Transformed item
+ */
+function transformItem(item) {
+    if (item.image_groups) {
+        const imageGroup = item.image_groups.find(function (i) {
+            return i.view_type === 'large'
+        }) || item.image_groups[0];
+        if (imageGroup) {
+            var firstImageInGroup = imageGroup.images[0];
+            item.image = firstImageInGroup
+        }
+    } else {
+        item.image = {
+            dis_base_link: algoliaData.noImages.large,
+            alt: item.name + ', large',
+        }
+    }
+
+    // adjusted price in user currency
+    if (item.promotionalPrice && item.promotionalPrice[algoliaData.currencyCode] !== null) {
+        item.promotionalPrice = item.promotionalPrice[algoliaData.currencyCode]
+    }
+
+    // price in user currency
+    if (item.price && item.price[algoliaData.currencyCode] !== null) {
+        item.price = item.price[algoliaData.currencyCode]
+    }
+
+    // currency symbol
+    item.currencySymbol = algoliaData.currencySymbol;
+
+
+    item.quickShowUrl = algoliaData.quickViewUrlBase + '?pid=' + (item.masterID || item.objectID);
+
+    // originating index
+    item.__indexName = algoliaData.productsIndex;
+
+    // url with queryID (used for analytics)
+    if (item.url) {
+        item.url = generateProductUrl({
+            objectID: item.objectID,
+            productUrl: item.url,
+            queryID: item.__queryID,
+            indexName: item.__indexName,
+        });
+    }
+
+    if (item.colorVariations) {
+        // Display the swatches only if at least one item has some colorVariations
+        displaySwatches = true;
+        item.colorVariations.forEach(colorVariation => {
+            colorVariation.variationURL = generateProductUrl({
+                objectID: item.objectID,
+                productUrl: colorVariation.variationURL,
+                queryID: item.__queryID,
+                indexName: item.__indexName,
+            });
+        });
+    }
+
+    // Master-level indexing
+    if (item.variants) {
+        let price;
+        item.variants.forEach(variant => {
+            price = variant.price[algoliaData.currencyCode]
+            variant.url = generateProductUrl({
+                objectID: item.objectID,
+                productUrl: variant.url,
+                queryID: item.__queryID,
+                indexName: item.__indexName,
+            });
+        });
+
+        // 1. Find the variant matching the selected facets, or use the default variant
+        let selectedVariant;
+        const sizeFacets = results._state.disjunctiveFacetsRefinements['variants.size'] || [];
```

Comment:
I created a function for handling it. I also removed this logic and used default variant directly because results are not available in recommend widgets. Please let me know if you have better idea.